### ISSUE TITLE
Lambda PutFunctionEventInvokeConfig fails with invalid response JSON, incorrect field name FunctionArn.

### ISSUE THREAD

**Issue by nhoughto** [localstack/localstack#3475]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
Lambda PutFunctionEventInvokeConfig operation fails on the response to terraform aws provider because the response doesn't match the schema, this causes terraform to keep retrying until failure
...

## Expected behavior
It works
...

## Actual behavior
It retries until failure
...

# Steps to reproduce
create a `aws_lambda_function_event_invoke_config` resource against localstack

The problem is that localstack is returning this response to the call,
```
{"DestinationConfig":{"OnFailure":{"Destination":"arn:aws:sqs:us-east-1:000000000000:local-events"},"OnSuccess":{"Destination":"arn:aws:sqs:us-east-1:000000000000:local-events"}},"FunctionArn":"arn:aws:lambda:us-east-1:000000000000:function:lambda-local","LastModified":"2021-01-15T06:01:48.944+0000","MaximumEventAgeInSeconds":null,"MaximumRetryAttempts":2}
```

But the spec says it should be FunctionName not FunctionArn - https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#PutFunctionEventInvokeConfigInput

**Comment by irahulranjan** [comment]
Hi @nhoughto, Please refer to the following documentation:
https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#PutFunctionEventInvokeConfigOutput
```
type PutFunctionEventInvokeConfigOutput struct {

    // A destination for events after they have been sent to a function for processing.
    //
    // Destinations
    //
    //    * Function - The Amazon Resource Name (ARN) of a Lambda function.
    //
    //    * Queue - The ARN of an SQS queue.
    //
    //    * Topic - The ARN of an SNS topic.
    //
    //    * Event Bus - The ARN of an Amazon EventBridge event bus.
    DestinationConfig *DestinationConfig `type:"structure"`

    // The Amazon Resource Name (ARN) of the function.
    FunctionArn *string `type:"string"`

    // The date and time that the configuration was last updated.
    LastModified *time.Time `type:"timestamp"`

    // The maximum age of a request that Lambda sends to a function for processing.
    MaximumEventAgeInSeconds *int64 `min:"60" type:"integer"`

    // The maximum number of times to retry when the function returns an error.
    MaximumRetryAttempts *int64 `type:"integer"`
    // contains filtered or unexported fields
}
```

**Comment by nhoughto** [comment]
lol ok sorry my hasty friday 5pm PR was a little off target!

heres the error
```
2021/01/15 06:01:48 [DEBUG] [aws-sdk-go] {"DestinationConfig":{"OnFailure":{"Destination":"arn:aws:sqs:us-east-1:000000000000:local-events"},"OnSuccess":{"Destination":"arn:aws:sqs:us-east-1:000000000000:local-events"}},"FunctionArn":"arn:aws:lambda:us-east-1:000000000000:function:core-lambda","LastModified":"2021-01-15T06:01:48.944+0000","MaximumEventAgeInSeconds":null,"MaximumRetryAttempts":2}
2021-01-15T06:01:48.950Z [DEBUG] plugin.terraform-provider-aws_v3.14.0_x5: 2021/01/15 06:01:48 [DEBUG] [aws-sdk-go] DEBUG: Unmarshal Response lambda/PutFunctionEventInvokeConfig failed, attempt 6/25, error SerializationError: failed decoding JSON RPC response
```

So maybe its the null `MaximumEventAgeInSeconds` that is causing the deser problem.

**Comment by irahulranjan** [comment]
Can you share a sample terraform template and commands to reproduce the issue?

**Comment by nhoughto** [comment]
`terraform init && terraform apply` i guess are the commands?

resource in question is:
```
resource "aws_lambda_function_event_invoke_config" "lambda_destination" {
  function_name = "function-name..."

  destination_config {
    on_failure {
      destination = "arn:aws:sqs:us-east-1:000000000000:local-events"
    }

    on_success {
      destination = "arn:aws:sqs:us-east-1:000000000000:local-events"
    }
  }
}
```

pretty standard, not much to it, I presume this has never been tested with terraform rather than anything too tricky going on.

**Comment by irahulranjan** [comment]
Hi @nhoughto,
which version of terraform are you using?

**Comment by nhoughto** [comment]
terraform 0.13.3, but the probably more relevant version is the aws provider, `3.14.0`. Few newer versions since then we could try..

**Comment by nhoughto** [comment]
Ok latest terraform aws provider still fails but it gives a more useful error
```
2021-01-17T02:17:43.458Z [INFO]  plugin.terraform-provider-aws_v3.24.1_x5: 2021/01/17 02:17:43 [DEBUG] [aws-sdk-go] {"DestinationConfig":{"OnFailure":{"Destination":"arn:aws:sqs:us-east-1:000000000000:local-events"},"OnSuccess":{"Destination":"arn:aws:sqs:us-east-1:000000000000:local-video-events"}},"FunctionArn":"arn:aws:lambda:us-east-1:000000000000:function:core-local","LastModified":"2021-01-17T02:17:43.451+0000","MaximumEventAgeInSeconds":null,"MaximumRetryAttempts":2}: timestamp=2021-01-17T02:17:43.458Z
2021-01-17T02:17:43.458Z [INFO]  plugin.terraform-provider-aws_v3.24.1_x5: 2021/01/17 02:17:43 [DEBUG] [aws-sdk-go] DEBUG: Unmarshal Response lambda/PutFunctionEventInvokeConfig failed, attempt 9/25, error SerializationError: failed decoding JSON RPC response
	status code: 200, request id: 
caused by: parsing time "2021-01-17T02:17:43.451+0000" as "2006-01-02T15:04:05.999999999Z": cannot parse "+0000" as "Z": timestamp=2021-01-17T02:17:43.458Z
```

notably: `caused by: parsing time "2021-01-17T02:17:43.451+0000" as "2006-01-02T15:04:05.999999999Z": cannot parse "+0000" as "Z": timestamp=2021-01-17T02:17:43.458Z`

**Comment by nhoughto** [comment]
Interestingly AWS itself appears to return quite a different value from LastModified, epoch time:

```
{
    "LastModified": 1573686021.479,
    "FunctionArn": "arn:aws:lambda:us-east-2:123456789012:function:my-function:$LATEST",
    "MaximumRetryAttempts": 0,
    "MaximumEventAgeInSeconds": 3600,
    "DestinationConfig": {
        "OnSuccess": {},
        "OnFailure": {}
    }
}
```
as per https://docs.aws.amazon.com/cli/latest/reference/lambda/put-function-event-invoke-config.html

**Comment by nhoughto** [comment]
Use of `isoformat_milliseconds(datetime.utcnow()) + '+0000'` instead of `float(time.mktime(datetime.utcnow().timetuple()))` like most of the other places in the code seems like the cause, don't know enough about it to say with confidence tho..

One of ~four calls: https://github.com/localstack/localstack/blob/master/localstack/services/awslambda/lambda_api.py#L960

**Comment by irahulranjan** [comment]
Hi @nhoughto, I am currently evaluating what to do. If we match AWS, terraform doesn't work. We have to remove +0000 to make it work. On the other hand making that change could lead to issues with others. I will keep you posted.

**Comment by nhoughto** [comment]
From my research the current behaviour doesn't match AWS? AWS returns `1573686021.479` and localstack currently returns `"2021-01-17T02:17:43.451+0000"`

**Comment by irahulranjan** [comment]
Yes, if I match AWS we still get the same error in terraform.

**Comment by nhoughto** [comment]
Hmm thats not my experience with latest versions of everything, patching like this it all works quite nicely:

```
sed -i "s|isoformat_milliseconds(datetime.utcnow()) + '+0000'|float(time.mktime(datetime.utcnow().timetuple()))|g" ./localstack/utils/aws/aws_models.py
sed -i "s|'LastModified': str(self.last_modified)|'LastModified': self.last_modified|g" ./localstack/utils/aws/aws_models.py
```

**Comment by irahulranjan** [comment]
Hi @nhoughto, apologies for the delayed response. I was trying out on terraform 0.13 so unix_time was failing. I have fixed for terraform 0.14. https://github.com/localstack/localstack/pull/3482 should resolve the issue.
Thanks.

**Comment by whummer** [comment]
Thanks for the update @nhoughto @irahulranjan . This should be fixed in the meantime - can you please pull the `latest` Docker image and let us know if the problem persists. Thanks
